packages/gatsby-theme-carbon/src/components/Code/Code.js

```diff
@@ -76,14 +81,14 @@ const Code = ({ children, className: classNameProp = '', path, src }) => {
           </div>
         )}
       </Highlight>
-      {hasMoreThanNineLines && (
+      {isOverFlowEnabled && shouldOverflow && (
         <button
           className={cx(styles.showMoreButton, {
             [styles.dark]: interiorTheme === 'dark',
           })}
-          onClick={() => setShouldShowMore(!shouldShowMore)}
+          onClick={() => setOverflowExpanded(!isOverflowExpanded)}
           type="button">
-          {shouldShowMore ? (
+          {isOverflowExpanded ? (
```

Comment:
I refactored these booleans because the code was confusing to me.

Before to me it read as: `If I should show more, render the component that says "Show Less"`

I made it a little more clear what each bool represented.